compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1938,10 +1943,75 @@ public void visit(BLangRecordDestructure recordDeStmt) {
             return;
         }
         typeChecker.checkExpr(recordDeStmt.expr, this.env);
+        validateBindingPatternForRecordDestructure(recordDeStmt.varRef.recordRefFields, recordDeStmt.expr.getBType(),
+                recordDeStmt.expr.pos);
         checkRecordVarRefEquivalency(recordDeStmt.pos, recordDeStmt.varRef, recordDeStmt.expr.getBType(),
                                      recordDeStmt.expr.pos);
     }
 
+    private void validateBindingPatternForVariableRecord(BLangRecordVariable recordVar,
```

Comment:
Do we need this validation separately? Doesn't org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter#defineVariableList handle this?

I tried with 

```ballerina
type Foo record {|
    record {
        int b?;
    }[1] a?;
|};

function fn(Foo f) {
    var {a: [{b}]} = f;
}
``` 

The diagnostic for `a` was added twice (when debugging).

Reply:
Yeah, will remove this logic and handle from `SymbolEnter`.